application/src/test/java/org/togetherjava/tjbot/jda/JdaTester.java

```diff
@@ -487,6 +495,110 @@ public JdaTester() {
         return ErrorResponseException.create(reason, new Response(null, -1, "", -1, Set.of()));
     }
 
+    /**
+     * Creates a MessageReceivedEvent
+     * <p>
+     * The method creates a MessageReceivedEvent from the input message and mock the message author
+     * from the input user.
+     *
+     * @param message the message to create an event for
+     * @param user the author from the message
+     * @return the created MessageReceivedEvent
+     */
+    public MessageReceivedEvent createMessageReceivedEvent(@NotNull Message message,
+            @NotNull User user) {
+        doReturn(user).when(message).getAuthor();
```

Comment:
This method requires that `message` is a spy or mock, which you did not document.

In fact, I find it confusing that it touches the message. This line of code shouldnt be part of this methods responsibility, please remove it.

Reply:
MessageBuilder doesnt work with new MessageReceivedEvent thats why i created a spy, but still i think it would be better if you could look into the JdaTester part, i try to close the other open conversations.